### PR TITLE
Add minItems validation to listeners array

### PR DESCRIPTION
**Pull request by stanlyDoge**
Signed-off-by: Stanislav Knot <sknot@redhat.com>

### Type of change
- Enhancement / new feature

### Description
Fixes https://github.com/strimzi/strimzi-kafka-operator/issues/3675
Listeners array should have at least one item. We can perform validation using `minItems`.

### Checklist
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Reference relevant issue(s) and close them after merging

### CODE REVIEW

**Review by scholzj — COMMENTED** [review]
Should we have some tests for it in the CRD geenrator?

File: crd-generator/src/main/java/io/strimzi/crdgenerator/annotations/MinimumItems.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.crdgenerator.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
```
Comment (tombentley) [inline crd-generator/src/main/java/io/strimzi/crdgenerator/annotations/MinimumItems.java:12]: In hindsight documenting the annotations on the CrdGenerator class level Javadoc was a mistake. I know you've documented this there, but I think we should probably document the annotation here too. (In another PR we could move the documentation to each of the annotations we define, and only document the jackson annotations on the CrdGenerator class level Javadoc).

**Review by tombentley — COMMENTED** [review]
Have you tested that the kube apiserver now rejects empty `listeners`?

**Review by scholzj — APPROVED** [review]

File: crd-generator/src/main/java/io/strimzi/crdgenerator/annotations/MinimumItems.java (hunk 1 above)
Comment (stanlyDoge) [reply crd-generator/src/main/java/io/strimzi/crdgenerator/annotations/MinimumItems.java:12]: `listeners: []` gives `spec.kafka.listeners in the body should have at least 1 items`
I am a little bit confused. Should I change doc in this PR or we will do some larger refactoring later?

Comment (tombentley) [reply crd-generator/src/main/java/io/strimzi/crdgenerator/annotations/MinimumItems.java:18]: Perfect! Another PR for the doc is fine.

**Review by tombentley — APPROVED** [review]

**Pull request merged**